.github/workflows/release.yml

```diff
@@ -0,0 +1,152 @@
+          path: "${{ github.workspace }}/${{ matrix.runtime[0] }}/${{ matrix.runtime[1] }}/target"
+          key: srtool-target-${{ matrix.runtime[0] }}-${{ matrix.runtime[1] }}-${{ github.sha }}
+          restore-keys: |
+            srtool-target-${{ matrix.runtime[0] }}-${{ matrix.runtime[1] }}-
+            srtool-target-${{ matrix.runtime[0] }}-
+
+      - name: Build ${{ matrix.runtime[1] }}
+        id: srtool_build
+        uses: chevdor/srtool-actions@v0.7.0
+        with:
+          chain: ${{ matrix.runtime[1] }}
+          runtime_dir: "${{ matrix.runtime[0] }}/${{ matrix.runtime[1] }}"
+
+      - name: Store ${{ matrix.runtime[1] }} srtool digest to disk
+        run: |
+          echo '${{ steps.srtool_build.outputs.json }}' | jq > ${{ matrix.runtime[1] }}_srtool_output.json
+
+      - name: Upload ${{ matrix.runtime[1] }} srtool json
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ matrix.runtime[1] }}-srtool-json
+          path: ${{ matrix.runtime[1] }}_srtool_output.json
+
+      - name: Upload  ${{ matrix.runtime[1] }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ matrix.runtime[1] }}
+          path: |
+            ${{ steps.srtool_build.outputs.wasm_compressed }}
+
+  publish-draft-release:
+    runs-on: ubuntu-latest
+    needs:
+      - build-runtimes
+    outputs:
+      release_url: ${{ steps.create-release.outputs.html_url }}
+      asset_upload_url: ${{ steps.create-release.outputs.upload_url }}
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v3
+
+      - name: Download srtool json output
+        uses: actions/download-artifact@v3
+
+      - name: Archive context output
+        uses: actions/upload-artifact@v3
+        with:
+          name: release-notes-context
+          path: |
+            **/*_srtool_output.json
+
+      - name: Add runtime info to changelog
+        run: |
+          C() { >>CHANGELOG echo $@; }
+          SRTOOL() { <$(find . -name '*_srtool_output.json' | head -n1) jq -r $1; }
+          C "## Runtime info"
+          C "*These runtimes were built with $(SRTOOL rustc) using $(SRTOOL gen)*  "
+          for JSON in $(find . -name '*_srtool_output.json' | sort -sr); do
+            WASM() { <${JSON} jq -r .runtimes.compact.subwasm.$1; }
+
+            C "### $(WASM core_version.specName)"
+            C '```'
+            C "🏋️ Runtime Size:           $(numfmt --to iec-i --format "%.2f" $(WASM size)) ($(WASM size) bytes)"
+            C "🔥 Core Version:           $(WASM core_version.specName)-$(WASM core_version.specVersion) $(WASM core_version.implName)-$(WASM core_version.implVersion).tx$(WASM core_version.transactionVersion).au$(WASM core_version.authoringVersion)"
+            C "🎁 Metadata version:       V$(WASM metadata_version)"
+            C "🗳️ Blake2-256 hash:        $(WASM blake2_256)"
+            C "📦 IPFS:                   $(WASM ipfs_hash)"
+            C '```'
+          done
```

Comment:
Why not using a template such as `tera` as we usually do ?

That will make it more readable/maintainable with less bash-ery.
That makes it also be more robust and easier to debug. 

The templates already exist:
- https://github.com/paritytech/polkadot/blob/master/scripts/ci/changelog/templates/runtimes.md.tera
- https://github.com/paritytech/polkadot/blob/master/scripts/ci/changelog/templates/runtime.md.tera

You may want to tweak a few things but that should get you very close.

Reply:
I want to keep this quite self-contained. Personally I think the gazillion scripts and templates and generator dependencies and such in the polkadot repo are a pain to overview & maintain.

I'm trying to ensure that the bash-ery in this is kept to a minimum, if this ends up being more spaghetti than the polkadot templating I'm reworking it.

Reply:
I've just cleaned up the bash templating a little, it's just as readable as the tera templates now IMO